crates/bevy_ecs/src/query/state.rs

```diff
@@ -1162,12 +1162,19 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
     }
 }
 
-/// An error that occurs when retrieving a specific [`Entity`]'s query result.
+/// An error that occurs when retrieving a specific [`Entity`]'s query result from [`QueryState`].
```

Comment:
```suggestion
/// An error that occurs when retrieving a specific [`Entity`]'s query result from [`Query`] or [`QueryState`].
```

Reply:
I agree it would be better, and I thought about doing it, but I didn't do it because:
- `Query` is not imported in state.rs and I was thinking importing it just for the doc was maybe unnecessary (and I think it throws a compiler warning?)
- `QuerySingleError` did reference `QueryState` but not `Query`

Reply:
I'd use the alternative syntax for [doc linking](https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html) to avoid the importing problem. Also imo I would also change QuerySingleError to reference Query also.

Reply:
I fixed that with 16d460a8713a240b7ca3d49d52c0c62bb3248a61.

Is that the way you wanted me to write it? I thought about it but wasn't sure it was the right thing to do since I never saw that anywhere in Bevy's documentation.

Notice I also fixed a typo (that wasn't mine it was already like that but I didn't notice it was there): `resulted` -> `result`, and I modified the references to `single` and `single_mut` in three ways:
- reference `get_single` and `get_single_mut` instead because it's actually them which throw this error, `single` and `single_mut` panic (again, that mistake is in the current doc but I didn't notice)
- remove "visual" reference to `Query`/`QueryState` to keep it "neutral"
- link to `Query` instead of `QueryState`